lighthouse-core/audits/byte-efficiency/large-javascript-libraries.js

```diff
@@ -0,0 +1,167 @@
+ */
+
+/**
+ * @fileoverview This audit checks a page for any large JS libraries with smaller alternatives.
+ * These libraries can be replaced with functionally equivalent, smaller ones.
+ */
+
+'use strict';
+
+/** @typedef {{name: string, gzip: number, repository: string}} BundlePhobiaLibrary */
+
+/** @type {Record<string, Record<string, BundlePhobiaLibrary>>} */
+const libStats = require('./bundlephobia-database.json');
+
+/** @type {Record<string, string[]>} */
+const librarySuggestions = require('./library-suggestions.js').suggestions;
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when no known unnecessarily large libraries are detected on the page.*/
+  title: 'Avoids unnecessarily large JavaScript libraries',
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when some known unnecessarily large libraries are detected on the page.*/
+  failureTitle: 'Replace unnecessarily large JavaScript libraries',
+  /** Description of a Lighthouse audit that tells the user why they should care about the large Javascript libraries that have better alternatives. This is displayed after a user expands the section to see more. No character length limits. */
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    ' [Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of large JavaScript libraries that could be replaced. */
+  name: 'Library',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargeJavascriptLibraries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: BundlePhobiaLibrary, suggestions: BundlePhobiaLibrary[]}>} */
+    const libraryPairings = [];
+    const detectedLibs = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    const seenLibraries = new Set();
+
+    for (const detectedLib of detectedLibs) {
+      if (!detectedLib.npm || !libStats[detectedLib.npm]) continue;
+      const suggestions = librarySuggestions[detectedLib.npm] || [];
```

Comment:
result is the same given the current contents of this loop operating on items in this array, but for mental parsing I see the `|| []` and immediately think "wait why do we want to execute this loop at all if we don't have any suggestions", does that make sense?

```suggestion
      const suggestions = librarySuggestions[detectedLib.npm];
      if (!suggestions) continue;
```